cmd/kubeadm/app/phases/controlplane/volumes.go

```diff
@@ -76,6 +77,11 @@ func getHostPathVolumesForTheControlPlane(cfg *kubeadmapi.MasterConfiguration) c
 		mounts.NewHostPathMount(kubeadmconstants.KubeControllerManager, flexvolumeDirVolumeName, flexvolumeDirVolumePath, flexvolumeDirVolumePath, false, &hostPathDirectoryOrCreate)
 	}
 
+	// Only append the --cloud-config volume mount if there is such a file
+	if _, err := os.Stat(DefaultCloudConfigPath); err == nil {
```

Comment:
Here it overlaps with https://github.com/kubernetes/kubernetes/pull/56535. That one takes more conditions into consideration. We can combine that one here and close that PR (or merge it separately and rebase here. It will be more operations). cc @luxas @dims

Reply:
I'm happy to let https://github.com/kubernetes/kubernetes/pull/56535 merge first :)